src/components/common/TmSessionHardware.vue

```diff
@@ -45,26 +57,59 @@ export default {
   components: { TmBtn, HardwareState },
   data: () => ({
     status: `connect`,
-    connectionError: null
+    connectionError: null,
+    address: null
   }),
+  computed: {
+    submitCaption() {
+      return {
+        connect: "Sign In",
+        detect: "Waiting for Ledger",
+        confirmAddress: "Confirming Address"
+      }[this.status]
+    }
+  },
   methods: {
     setState(value) {
       this.$store.commit(`setSessionModalView`, value)
     },
-    setStatus(value) {
-      this.status = value
-    },
-    setConnectionError(error) {
-      this.connectionError = error
+    async signIn() {
+      this.connectionError = null
+      this.status = `detect`
+      this.address = null
+      try {
+        this.address = await this.$store.dispatch(`connectLedgerApp`)
+      } catch ({ message }) {
+        this.status = `connect`
+        this.connectionError = message
+        return
+      }
+
+      this.status = `confirmAddress`
+      if (await this.confirmAddress()) {
+        await this.$store.dispatch(`signIn`, {
+          sessionType: `ledger`,
+          address: this.address
+        })
+        return
+      }
+
+      this.status = `connect`
     },
-    async connectLedger() {
-      this.setStatus(`detect`)
+    async confirmAddress() {
       try {
-        await this.$store.dispatch(`connectLedgerApp`)
-      } catch (error) {
-        this.setStatus(`connect`)
-        this.setConnectionError(error.message)
+        await this.$store.dispatch("confirmLedgerAddress")
+        return true
+      } catch ({ message }) {
+        switch (message) {
+          case `Transaction rejected`:
+            this.connectionError = `Account address rejected`
```

Comment:
why would an address get rejected?

Reply:
The address approval is associated with an action, where you can "reject" the address.